Analysis/Tasks/PWGHF/HFMCMatchingValidation.cxx

```diff
@@ -0,0 +1,200 @@
+    registry.fill(HIST("hMCEvt_countAverageBbar"), bbarpercollision);
+    registry.fill(HIST("hMCEvt_CouterPerCollision_Dplus"), counter[0]);
+    registry.fill(HIST("hMCEvt_CouterPerCollision_Dstar"), counter[1]);
+    registry.fill(HIST("hMCEvt_CouterPerCollision_Dzero"), counter[2]);
+    registry.fill(HIST("hMCEvt_CouterPerCollision_LambdaC"), counter[3]);
+  }
+};
+
+//Task Reco Level
+struct TaskRecoLevel {
+  HistogramRegistry registry{
+    "registry",
+    {{"hist_pT", "PT difference reco with MC", {HistType::kTH1F, {{2000, -1, 1}}}},
+     {"hist_pxConservation_Reco", "Px conservation Reco Candidates (MC matched)", {HistType::kTH1F, {{2000, -1, 1}}}},
+     {"hist_pyConservation_Reco", "Py conservation Reco Candidates (MC matched)", {HistType::kTH1F, {{2000, -1, 1}}}},
+     {"hist_pzConservation_Reco", "Pz conservation Reco Candidates (MC matched)", {HistType::kTH1F, {{2000, -1, 1}}}},
+     {"hist_pConservation_Reco", "P  conservation Reco Candidates (MC matched)", {HistType::kTH1F, {{2000, -1, 1}}}},
+     {"hist_SecVx", "Sec. Vertex (x) difference reco with MC true (MC matched)", {HistType::kTH1F, {{200, -1, 1}}}},
+     {"hist_SecVy", "Sec. Vertex (y) difference reco with MC true (MC matched)", {HistType::kTH1F, {{200, -1, 1}}}},
+     {"hist_SecVz", "Sec. Vertex (z) difference reco with MC true (MC matched)", {HistType::kTH1F, {{200, -1, 1}}}},
+     {"hist_decLen", "DecayLenght difference reco with MC true (MC matched)", {HistType::kTH1F, {{200, -1, 1}}}}}};
+
+  Configurable<int> d_selectionFlagD0{"d_selectionFlagD0", 0, "Selection Flag for D0"};
+  Configurable<int> d_selectionFlagD0bar{"d_selectionFlagD0bar", 0, "Selection Flag for D0bar"};
+
+  Filter filterSelectCandidates = (aod::hf_selcandidate_d0::isSelD0 >= d_selectionFlagD0 || aod::hf_selcandidate_d0::isSelD0bar >= d_selectionFlagD0bar);
+
+  void process(soa::Filtered<soa::Join<aod::HfCandProng2, aod::HFSelD0Candidate, aod::HfCandProng2MCRec>> const& candidates, aod::BigTracksMC const& tracks, aod::McParticles const& particlesMC)
+  {
+    int indexParticle = 0;
+    double pxDiff, pyDiff, pzDiff, p_magnitude_diff;
+    double decayLength;
+    for (auto& candidate : candidates) {
+      if (!(candidate.hfflag() & 1 << D0ToPiK))
+        continue;
+      if (std::abs(candidate.flagMCMatchRec()) == 1 << D0ToPiK) {
+        indexParticle = RecoDecay::getMother(particlesMC, candidate.index0_as<aod::BigTracksMC>().label(), 421);
+        if (indexParticle < 0)
+          continue;
+        auto mother = particlesMC.iteratorAt(indexParticle);
+        registry.fill(HIST("hist_pT"), candidate.pt() - mother.pt());
+        //Check the momentum conservation Reco candidates with their daughters
+        pxDiff = candidate.px() - candidate.pxProng0() - candidate.pxProng1();
+        pyDiff = candidate.py() - candidate.pyProng0() - candidate.pyProng1();
+        pzDiff = candidate.pz() - candidate.pzProng0() - candidate.pzProng1();
```

Comment:
These have to be identically equal zero by construction because the candidate momentum components are defined this way.